launch/client.py

```diff
@@ -744,20 +765,63 @@ def get_async_response(self, async_task_id: str) -> Dict[str, Any]:
         )
         return resp
 
-    def batch_async_request(self, endpoint_id: str, urls: List[str]):
+    def batch_async_request(
+        self,
+        bundle_name: str,
+        urls: List[str],
+        batch_url_file_location: str = None,
+        serialization_format: str = "json",
+    ):
         """
         Sends a batch inference request to the Model Endpoint at endpoint_id, returns a key that can be used to retrieve
         the results of inference at a later time.
 
         Parameters:
-            endpoint_id: The id of the endpoint to make the request to
+            bundle_name: The id of the bundle to make the request to
+            serialization_format: Serialization format of output, either 'pickle' or 'json'.
+                'pickle' corresponds to pickling results + returning
             urls: A list of urls, each pointing to a file containing model input.
                 Must be accessible by Scale Launch, hence urls need to either be public or signedURLs.
+            batch_url_file_location: In self-hosted mode, the input to the batch job will be uploaded
+                to this location if provided. Otherwise, one will be determined from bundle_location_fn()
 
         Returns:
             An id/key that can be used to fetch inference results at a later time
         """
-        raise NotImplementedError
+        f = StringIO()
+        make_batch_input_file(urls, f)
+        f.seek(0)
+
+        if self.self_hosted:
+            # TODO make this not use bundle_location_fn()
+            if batch_url_file_location is None:
+                file_location = self.bundle_location_fn()  # type: ignore
+            else:
+                file_location = batch_url_file_location
+            self.upload_batch_csv_fn(  # type: ignore
+                f.getvalue(), file_location
+            )
+        else:
+            # TODO make this not use MODEL_BUNDLE_SIGNED_URL_PATH
```

Comment:
This needs a change on the scaleapi side to accomplish this

Reply:
IIUC `model_bundle_s3_url['key']` will have a prefix "model_bundles"?

Reply:
oh wait lemme change this lmao, I added the scaleapi change

currently the prefix will be `model_bundles`, once I make the change it'll be something like `batch_job_inputs`